src/sagas/CreateClaim.saga.ts

```diff
@@ -0,0 +1,52 @@
+import { Frost } from '@po.et/frost-client'
+import { Actions } from 'actions'
+import { delay, SagaIterator } from 'redux-saga'
+import { call, takeLatest, put, ForkEffect } from 'redux-saga/effects'
+const { toast } = require('react-toastify')
+
+import { Configuration } from 'configuration'
+
+const FrostClient = new Frost({ host: Configuration.frostApiUrl })
+
+export function CreateClaimSaga(): () => IterableIterator<ForkEffect> {
+  return function*(): IterableIterator<ForkEffect> {
+    yield takeLatest(Actions.CreateClaim.CREATE_CLAIM, handleOnCreateClaim)
+    yield takeLatest(Actions.CreateClaim.CREATE_CLAIM_SUCCESS, handleOnCreateClaimSuccess)
+    yield takeLatest(Actions.CreateClaim.CREATE_CLAIM_ERROR, handleOnCreateClaimError)
+  }
+}
+
+export function* handleOnCreateClaim(action: any): SagaIterator {
+  try {
+    const { token, work } = action.payload
+    yield put(Actions.LoadingPage.onLoadingOn())
+    const { workId } = yield call([FrostClient, FrostClient.createWork], token, work)
+    yield put(Actions.LoadingPage.onLoadingFull())
+    yield put(Actions.CreateClaim.onCreateClaimSuccess({ workId }))
+  } catch (error) {
+    if (error.message) yield put(Actions.CreateClaim.onCreateClaimError(error.message))
+    else yield put(Actions.CreateClaim.onCreateClaimError(error))
+
+  }
+}
+
+export function* handleOnCreateClaimSuccess({ payload: { workId } }: any): SagaIterator {
+  yield put(Actions.NotificationBar.onShowNotificationBar({
+    type: 'link-success',
+    message: `${Configuration.mainExplorerUrl}/works/${workId}`,
+  }))
+  yield call(delay, 3000)
+  yield put(Actions.NotificationBar.onHideNotificationBar())
+  yield call(delay, 2000)
+  yield put(Actions.NotificationBar.onResetNotificationBar())
+}
+
+export function* handleOnCreateClaimError({ payload: error }: any): SagaIterator {
+  yield put(Actions.LoadingPage.onLoadingFull())
+  yield call(toast.error, error, {
+    className: 'toast',
+    autoClose: 2500,
+  })
+  yield call(delay, 300)
```

Comment:
Seems a bit odd that we wipe this error after 300ms. Are we not using that data anywhere to show the user, if not, then why even store it in the reducer? If this is a pattern of the app again, just ignore this, for now, its something that can be revisited later.

Reply:
The error is displayed in the toast message and it closes after 2500ms. It is a pattern throughout the app to clear the message after the 300ms. It still shows for the user for the full 2500ms though.

Reply:
Yeah, I see that the error is shown via toast, but it does not sound like the data we store in the reducer is ever used, its stored for 300ms and then cleared, doesn't make much sense, but if its part of how the app pattern at the moment then that's fine.

Reply:
It will be another thing to make sure we fix in the new app.

Reply:
I wouldn't worry too much about following the current architecture to the letter in frost-web since we're gonna EOL it as a stand-alone application soon... could be a good place to experiment with new approaches that we can later use in the new explorer

Reply:
> It will be another thing to make sure we fix in the new app.

Maybe it'd be nice to take notes about things we find in the old architecture of frost-web and explorer-web and don't like in the random repo. 

It'd be nice to start the new project the other way round, sort of doing RDD on the whole architecture, for at least to have a documented architecture from day one.